src/tss2-fapi/ifapi_helpers.c

```diff
@@ -1870,6 +1870,135 @@ ifapi_get_nv_start_index(const char *path, TPM2_HANDLE *start_nv_index)
     return_error2(TSS2_FAPI_RC_BAD_PATH, "Invalid NV path: %s", path);
 }
 
+/** Check whether NV index is appropriate for NV path.
+ *
+ * The value will be checked  based on e TCG handle registry.
+ *
+ * @param[in]  path The path used for the NV object.
+ * @param[out] nv_index The NV index to be used.
+ *
+ * @retval TSS2_RC_SUCCESS If the index for the path can be determined.
+ * @retval TSS2_FAPI_RC_BAD_PATH If the path is not valid.
+ * @retval TSS2_FAPI_RC_BAD_VALUE If the nv index is not appropriate for the path.
+ * @retval TSS2_FAPI_RC_MEMORY if not enough memory can be allocated.
+ */
+TSS2_RC
+ifapi_check_nv_index(const char *path, TPM2_HANDLE nv_index)
+{
+    TSS2_RC r = TSS2_RC_SUCCESS;
+    NODE_STR_T *dir_list = split_string(path, IFAPI_FILE_DELIM);
+
+    return_if_null(dir_list, "Out of memory.", TSS2_FAPI_RC_MEMORY);
+    if (dir_list->next && strcmp(dir_list->str, "nv") == 0) {
+        if (strcmp(dir_list->next->str, "TPM") == 0) {
+            if (nv_index >= 0x01000000 && nv_index <= 0x013fffff)
+                goto cleanup;
+            goto_error(r, TSS2_FAPI_RC_BAD_VALUE,
```

Comment:
This goto cleanup; followed by the goto_error(), which goes to the same label looks a bit strange. We should probably reverse the logic and just do:
if (nv_index < 0x01000000 || nv_index > 0x013fffff)
    goto_error(r, TSS2_FAPI_RC_BAD_VALUE, ...)
...

and get remove the goto cleanup;